### PR TITLE
Digitdustin/lesson backend integration

### PR DESCRIPTION
**Pull request by digitdustin**
Introduces integrations with the backend for lessons.

# Description

PR is currently a WIP and up for incremental review.

Includes:
- Admin frontend to create lessons
- Ability to fetch existing lessons and edit their content
- Submission of lessons to backend via redux
- Display of lessons on student frontend
- Transition of all admin-produced content (lessons/problems) to '/playground' route
- Multiple choice and multiple select creation and improved admin creation modal
- Merge fetch and navigation of content per modules (on admin/modules and /playground)
- Updated admin lessons editor toolbar and styling
- etc.

### CODE REVIEW

File: src/shared/constants.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export const baseAPIURL = "https://edugator-admin.com/";
+export const baseAPIURL = "http://localhost:8080/"; //https://edugator-admin.com/";
```
Comment (marcgabe15) [inline src/shared/constants.ts:1]: this will need to change back to edugator-admin

File: src/App.tsx [hunk 2]
```diff
@@ -45,12 +59,12 @@ function App() {
             <PrivateRoute exact path={Routes.AdminCodeWithProblem}>
               <CodeEditorPage />
             </PrivateRoute>
-            <Route path={Routes.CodeWithProblem}>
+            {/* <Route path={Routes.CodeWithProblem}>
```
Comment (marcgabe15) [inline src/App.tsx:62]: remove comment

File: src/shared/TopicSidebar.tsx [hunk 3]
```diff
@@ -13,6 +13,8 @@ import {
   MathOperations,
   House,
   ListBullets,
+  BookOpen,
```
Comment (marcgabe15) [inline src/shared/TopicSidebar.tsx:16]: I see this got imported but where is it being used?

**Review by marcgabe15 — CHANGES_REQUESTED** [review]

File: src/pages/Playground/PlaygroundContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        })
+      );
+    } else if (params && params["lessonId"]) {
+      setContentType("lesson");
+      dispatch(
+        requestLesson({
+          lessonId: params["lessonId"],
+        })
+      );
+      console.log("lessonId", params["lessonId"]);
```
Comment (marcgabe15) [inline src/pages/Playground/PlaygroundContainer.tsx:85]: remove console log

File: src/pages/LearnPage/LearnPage.tsx [hunk 5]
```diff
@@ -100,124 +164,136 @@ function LearnPage() {
...
-                      caption={block.content.caption}
-                      height={block.content.height}
-                      width={block.content.width}
-                      alignment={block.content.alignment}
+                return (
+                  <div key={i} style={{ width: "100%" }}>
+                    <Markup
+                      transform={transform}
+                      className="inter"
+                      content={JSON.parse(JSON.stringify(block.content.html))}
```
Comment (marcgabe15) [inline src/pages/LearnPage/LearnPage.tsx:251]: json.parse and json stringify can throw can we have a try catch wrapped around this?

File: src/pages/ContentEditor/contentEditorPageSlice.ts [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+      state.fetchingContent = false;
+
+      state.metadata = {
+        title: action.payload.title,
+        author: action.payload.author,
+      };
+      state.contentEditor = {
+        content: { ...action.payload.content },
+        editableContent: {
+          // @ts-ignore
```
Comment (marcgabe15) [inline src/pages/ContentEditor/contentEditorPageSlice.ts:153]: what is this ignoring?

File: src/pages/Playground/PlaygroundContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState } from "react";
```
Comment (marcgabe15) [inline src/pages/Playground/PlaygroundContainer.tsx:1]: so everything from CodeEditor just moved here?

File: src/pages/ContentEditor/components/form.tsx [hunk 8]
```diff
@@ -1,81 +1,142 @@
...
+    let contentArr: any[] = [];
+    pageJsonData.title = JSON.stringify(title, undefined, 2);
+    pageJsonData.author = JSON.stringify(author, undefined, 2);
+    jsonData.forEach((content: any) => {
+      console.log(content);
+      contentArr.push(content);
     });
-  };
+    pageJsonData.content = contentArr;
+    // let exportData = pageJsonData.content.join(",\n");
```
Comment (marcgabe15) [inline src/pages/ContentEditor/components/form.tsx:75]: remove comment

**Comment by marcgabe15** [comment]
run prettier as well

File: src/pages/Playground/PlaygroundContainer.tsx (hunk 7 above)
Comment (digitdustin) [reply src/pages/Playground/PlaygroundContainer.tsx:1]: The code from CodeEditor that just involved page layout (sidebar, header) was moved to here, and content that's inside main page area was compartmentalized to CodeEditorPage for problems and LearnPage for lessons

File: src/pages/ContentEditor/contentEditorPageSlice.ts (hunk 6 above)
Comment (digitdustin) [reply src/pages/ContentEditor/contentEditorPageSlice.ts:153]: Types with lesson payload, fixed

**Pull request closed without merging**